### PR TITLE
Clean + upgrade for the Create Trip and Overview Screen 

### PR DESCRIPTION
**Pull request by Exino329**
In this PR, the following updates have been made :

1. The description not single lined anymore (it was a small visual bug fix) And the size of the title is now limited to 35 caracters).
2. The overview screen is upgraded : Now there is a button for each trip at the right allowing someone to share the Trip to friends (using the trip id) via email or other app (such as whats app for example).
3. The CreateTripViewModel has been merged in the OverviewViewModel.
4. Padding was adjusted in the OverviewViewModel to make the display cleaner when scrollable column showing the trips is full.

### CODE REVIEW

File: app/src/main/java/com/github/se/wanderpals/ui/screens/overview/OverviewTrip.kt
Comment (annhit29) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/overview/OverviewTrip.kt:122]: It's good, one thing that can be improved is: there's a background color (let's say grey) for the share icon when clicking on it so that the user will not be confused about which thing (the box button for going to the dashboard or the icon button for sharing the trip) he/she is clicking

File: app/src/main/java/com/github/se/wanderpals/ui/screens/CreateTrip.kt
Comment (annhit29) [inline app/src/main/java/com/github/se/wanderpals/ui/screens/CreateTrip.kt:134]: can create an else case for the case when newTitle.length > MAX_TITLE_LENGTH

**Review by annhit29 — CHANGES_REQUESTED** [review]
All tests have passed on my side.
Two things can be improved: 
1. for your first point: when the title size exceeds 35 characters, there can be a red text displayed to say "at most 35 characters" to be clear (also see the second review message)
2. for your second point: see the first review message

The coverage is 76.1%, almost 80%!

Good job :D

File: app/src/main/java/com/github/se/wanderpals/ui/screens/CreateTrip.kt
Comment (Exino329) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/CreateTrip.kt:134]: Here an else is not necessary since we just want to limit the maximum possible size, (i.e. if the title length is higher than 35, don't update)

File: app/src/main/java/com/github/se/wanderpals/ui/screens/overview/OverviewTrip.kt
Comment (Exino329) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/overview/OverviewTrip.kt]: Nice catch ! I did not thought about this small detail.

File: app/src/main/java/com/github/se/wanderpals/ui/screens/CreateTrip.kt
Comment (Exino329) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/CreateTrip.kt:134]: Also note that there is no need to advertise the user that he has reached the maximum length for the title. Since there is already a little text showing the title length as a fraction

**Review by annhit29 — APPROVED** [review]
I tried and tested on my emulator, everything is good now!
And for you final version (just before merging), the line coverage is 90.4% :D

File: app/src/main/java/com/github/se/wanderpals/ui/screens/CreateTrip.kt
Comment (annhit29) [reply app/src/main/java/com/github/se/wanderpals/ui/screens/CreateTrip.kt:134]: Yes, I didn't see the character count `"${name.length}/$MAX_TITLE_LENGTH"` with `MAX_TITLE_LENGTH = 35`, because it was a bit low i.e. it was much closer to the Budger field. We (one of the team members) can move it a bit higher to the Title field later :D

**Pull request merged**
